quinn-proto/src/cid_generator.rs

```diff
@@ -7,17 +7,18 @@ use crate::MAX_CID_SIZE;
 
 /// Generates connection IDs for incoming connections
 pub trait ConnectionIdGenerator: Send {
-    /// Generates a new CID with finite lifetime
+    /// Generates a new CID
     ///
     /// Connection IDs MUST NOT contain any information that can be used by
     /// an external observer (that is, one that does not cooperate with the
     /// issuer) to correlate them with other connection IDs for the same
     /// connection.
-    ///
-    /// In the future, connection IDs will be retired after the returned `Duration`, if any.
-    fn generate_cid(&mut self) -> (ConnectionId, Option<Duration>);
+    fn generate_cid(&mut self) -> ConnectionId;
     /// Returns the length of a CID for cononections created by this generator
     fn cid_len(&self) -> usize;
+    /// Returns the lifetime of generated Connection IDs
+    /// Connection IDs will be retired after the returned `Duration`, if any
+    fn cid_lifetime(&self) -> Option<Duration>;
```

Comment:
I'm not sure whether it makes things easier or not. But another idea to express "no lifetime" is to pick a very high or maximum duration (like [`Duration::MAX`](https://doc.rust-lang.org/std/time/struct.Duration.html#associatedconstant.MAX) - which is unfortunately not stable), and use this instead of `Option<Duration>`.

This can avoid having to deal with `Option`s everywhere.
Obviously requires all operations to properly use saturating adds.

Reply:
I find it a bit of a mixed bag, as I feel like the `Option` makes the intent more clear than having a sentinel `Duration` value.

Reply:
I agree with Dirkjan on this topic that a `Option` shows the intention clearer.